wallet/src/backend.rs

```diff
@@ -137,64 +169,154 @@ impl<'a, Meta: Serialize + DeserializeOwned + Send> WalletBackend<'a, CapeLedger
-        // contract above succeeded, and the mock EQS is supposed to be tracking the contract state.
-        let mut mock_eqs = self.mock_eqs.lock().await;
-        mock_eqs.network().store_call_data(
-            info.uid.unwrap_or_else(|| TransactionUID(txn.hash())),
-            info.memos.into_iter().flatten().collect(),
-            info.sig,
-        );
-        mock_eqs.submit(txn).unwrap();
-
-        Ok(())
     }
 
-    ////////////////////////////////////////////////////////////////////////////////////////////////
-    // The remaining backend methods are still mocked, pending completion of the EQS
-    //
-
     async fn create(&mut self) -> Result<WalletState<'a, CapeLedger>, CapeWalletError> {
-        let state = self
-            .mock_eqs
-            .lock()
-            .await
-            .network()
-            .create_wallet(self.universal_param)?;
-        self.storage().await.create(&state).await?;
-        Ok(state)
+        let state: CapState = self.get_eqs("get_cap_state").await?;
+
+        // `records` should be _almost_ completely sparse. However, even a fully pruned Merkle tree
+        // contains the last leaf appended, but as a new wallet, we don't care about _any_ of the
+        // leaves, so make a note to forget the last one once more leaves have been appended.
+        let record_mt = MerkleTree::restore_from_frontier(
+            state.ledger.record_merkle_commitment,
+            &state.ledger.record_merkle_frontier,
+        )
+        .ok_or_else(|| CapeWalletError::Failed {
+            msg: String::from("cannot reconstruct Merkle tree from frontier"),
+        })?;
+        let merkle_leaf_to_forget = if record_mt.num_leaves() > 0 {
+            Some(record_mt.num_leaves() - 1)
+        } else {
+            None
+        };
+
+        Ok(WalletState {
+            proving_keys: Arc::new(gen_proving_keys(self.universal_param)),
+            txn_state: TransactionState {
+                validator: CapeTruster::new(state.ledger.state_number, record_mt.num_leaves()),
+                now: EventIndex::from_source(EventSource::QueryService, state.num_events as usize),
+                // Completely sparse nullifier set
+                nullifiers: Default::default(),
+                record_mt,
+                records: RecordDatabase::default(),
+                merkle_leaf_to_forget,
+                transactions: Default::default(),
+            },
+            key_scans: Default::default(),
+            key_state: Default::default(),
+            assets: Default::default(),
+            viewing_accounts: Default::default(),
+            freezing_accounts: Default::default(),
+            sending_accounts: Default::default(),
+        })
     }
 
     async fn subscribe(&self, from: EventIndex, to: Option<EventIndex>) -> Self::EventStream {
-        self.mock_eqs.lock().await.network().subscribe(from, to)
+        // Sleep at least 500ms between each request. This should try to match the EQS polling
```

Comment:
The EQS polling frequency is configurable... and may eventually be replaced with the ethers subscribe mechanism. I'm not saying capping this is a bad idea, but maybe something that can be configured, in case this is embedded or otherwise invoked in circumstances where half a second is too big a delay?